src/LibraryManager/Providers/Unpkg/UnpkgCatalog.cs

```diff
@@ -194,16 +193,16 @@ private void GetFiles(JObject fileObject, List<string> files)
                 // library version completion
                 else
                 {
-                    completionSet.Start = unpkgLibraryId.Name.Length + 1;
-                    completionSet.Length = unpkgLibraryId.Version.Length;
+                    completionSet.Start = name.Length + 1;
+                    completionSet.Length = version.Length;
 
-                    NpmPackageInfo npmPackageInfo = await NpmPackageInfoCache.GetPackageInfoAsync(unpkgLibraryId.Name, CancellationToken.None);
-                    foreach (SemanticVersion version in npmPackageInfo.Versions)
+                    NpmPackageInfo npmPackageInfo = await NpmPackageInfoCache.GetPackageInfoAsync(name, CancellationToken.None);
+                    foreach (SemanticVersion semVersion in npmPackageInfo.Versions)
                     {
                         CompletionItem completionItem = new CompletionItem
                         {
-                            DisplayText = unpkgLibraryId.Name + "@" + version.ToString(),
-                            InsertionText = unpkgLibraryId.Name + "@" + version.ToString()
+                            DisplayText = name + "@" + semVersion.ToString(),
```

Comment:
>name + "@" + semVersion.ToString() [](start = 42, length = 34)

No need to do this twice

Reply:
Which one should I remove? I am not familiar with the completion code... Just renamed the variables : ) 

---
In reply to: [195510285](https://github.com/aspnet/LibraryManager/pull/164#discussion_r195510285) [](ancestors = 195510285)

Reply:
string itemText = name + "@" + semVersion.ToString();
CompletionItem completionItem = new CompletionItem
{
    DisplayText = itemText,
    InsertionText = itemText
};

---
In reply to: [195540212](https://github.com/aspnet/LibraryManager/pull/164#discussion_r195540212) [](ancestors = 195540212,195510285)